flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/operators/runtimefilter/GlobalRuntimeFilterBuilderOperatorTest.java

```diff
@@ -0,0 +1,168 @@
+            assertThat(globalBloomFilter.testHash("var7".hashCode())).isTrue();
+            assertThat(globalBloomFilter.testHash("var8".hashCode())).isTrue();
+            assertThat(globalBloomFilter.testHash("var9".hashCode())).isTrue();
+            assertThat(globalBloomFilter.testHash("var10".hashCode())).isTrue();
+        }
+    }
+
+    /**
+     * Test the case that all input local runtime filters are normal, but the merged global filter
+     * is over-max-row-count.
+     */
+    @Test
+    void testNormalInputAndOverMaxRowCountOutput() throws Exception {
+        try (StreamTaskMailboxTestHarness<RowData> testHarness =
+                createGlobalRuntimeFilterBuilderOperatorHarness(9)) {
+            // process elements
+            testHarness.processElement(
+                    new StreamRecord<RowData>(
+                            GenericRowData.of(5, BloomFilter.toBytes(createBloomFilter1()))));
+            testHarness.processElement(
+                    new StreamRecord<RowData>(
+                            GenericRowData.of(5, BloomFilter.toBytes(createBloomFilter2()))));
+            testHarness.processEvent(new EndOfData(StopMode.DRAIN), 0);
+
+            // test the output
+            Queue<Object> outputs = testHarness.getOutput();
+            assertThat(outputs.size()).isEqualTo(1);
+
+            RowData outputRowData = ((StreamRecord<RowData>) outputs.poll()).getValue();
+            assertThat(outputRowData.getArity()).isEqualTo(2);
+
+            int globalCount = outputRowData.getInt(0);
+            assertThat(globalCount).isEqualTo(OVER_MAX_ROW_COUNT);
+            assertThat(outputRowData.isNullAt(1)).isTrue();
+        }
+    }
+
+    /** Test the case that one of the input local runtime filters is over-max-row-count. */
+    @Test
+    void testOverMaxRowCountInput() throws Exception {
+        try (StreamTaskMailboxTestHarness<RowData> testHarness =
+                createGlobalRuntimeFilterBuilderOperatorHarness(10)) {
+            // process elements
+            testHarness.processElement(
+                    new StreamRecord<RowData>(
+                            GenericRowData.of(5, BloomFilter.toBytes(createBloomFilter1()))));
+            testHarness.processElement(
+                    new StreamRecord<RowData>(GenericRowData.of(OVER_MAX_ROW_COUNT, null)));
+            testHarness.processEvent(new EndOfData(StopMode.DRAIN), 0);
+
+            // test the output
+            Queue<Object> outputs = testHarness.getOutput();
+            assertThat(outputs.size()).isEqualTo(1);
+
+            RowData outputRowData = ((StreamRecord<RowData>) outputs.poll()).getValue();
+            assertThat(outputRowData.getArity()).isEqualTo(2);
+
+            int globalCount = outputRowData.getInt(0);
+            assertThat(globalCount).isEqualTo(OVER_MAX_ROW_COUNT);
+            assertThat(outputRowData.isNullAt(1)).isTrue();
+        }
+    }
+
+    private static BloomFilter createBloomFilter1() {
```

Comment:
It would be better if we can extract a util method
```
    private static BloomFilter createBloomFilter(int rowCount, int offset) {
        final BloomFilter bloomFilter = RuntimeFilterUtils.createOnHeapBloomFilter(rowCount, 0.05);
        for (int i = 0; i < rowCount; i++) {
            bloomFilter.addHash(String.format("var%s", offset + i).hashCode());
        }
        return bloomFilter;
    }
```